tests/boards/nrf52_bsim/egu/src/main.c

```diff
@@ -0,0 +1,505 @@
+						NRF_EGU_EVENT_TRIGGERED15),
+			  "NRF_EGU_0 incorrect address of event trigger 15");
+
+	zassert_equal((size_t)&NRF_EGU1->EVENTS_TRIGGERED[0],
+			  nrf_egu_event_address_get(NRF_EGU1, NRF_EGU_EVENT_TRIGGERED0),
+			  "NRF_EGU_1 incorrect address of event trigger 0");
+	zassert_equal((size_t)&NRF_EGU1->EVENTS_TRIGGERED[1],
+			  nrf_egu_event_address_get(NRF_EGU1, NRF_EGU_EVENT_TRIGGERED1),
+			  "NRF_EGU_1 incorrect address of event trigger 1");
+	zassert_equal((size_t)&NRF_EGU1->EVENTS_TRIGGERED[15],
+			  nrf_egu_event_address_get(NRF_EGU1,
+						NRF_EGU_EVENT_TRIGGERED15),
+			  "NRF_EGU_1 incorrect address of event trigger 15");
+
+	zassert_equal((size_t)&NRF_EGU2->EVENTS_TRIGGERED[0],
+			  nrf_egu_event_address_get(NRF_EGU2, NRF_EGU_EVENT_TRIGGERED0),
+			  "NRF_EGU_2 incorrect address of event trigger 0");
+	zassert_equal((size_t)&NRF_EGU2->EVENTS_TRIGGERED[1],
+			  nrf_egu_event_address_get(NRF_EGU2, NRF_EGU_EVENT_TRIGGERED1),
+			  "NRF_EGU_2 incorrect address of event trigger 1");
+	zassert_equal((size_t)&NRF_EGU2->EVENTS_TRIGGERED[15],
+			  nrf_egu_event_address_get(NRF_EGU2,
+						NRF_EGU_EVENT_TRIGGERED15),
+			  "NRF_EGU_2 incorrect address of event trigger 15");
+
+	zassert_equal((size_t)&NRF_EGU3->EVENTS_TRIGGERED[0],
+			  nrf_egu_event_address_get(NRF_EGU3, NRF_EGU_EVENT_TRIGGERED0),
+			  "NRF_EGU_3 incorrect address of event trigger 0");
+	zassert_equal((size_t)&NRF_EGU3->EVENTS_TRIGGERED[1],
+			  nrf_egu_event_address_get(NRF_EGU3, NRF_EGU_EVENT_TRIGGERED1),
+			  "NRF_EGU_3 incorrect address of event trigger 1");
+	zassert_equal((size_t)&NRF_EGU3->EVENTS_TRIGGERED[15],
+			  nrf_egu_event_address_get(NRF_EGU3,
+						NRF_EGU_EVENT_TRIGGERED15),
+			  "NRF_EGU_3 incorrect address of event trigger 15");
+
+	zassert_equal((size_t)&NRF_EGU4->EVENTS_TRIGGERED[0],
+			  nrf_egu_event_address_get(NRF_EGU4, NRF_EGU_EVENT_TRIGGERED0),
+			  "NRF_EGU_4 incorrect address of event trigger 0");
+	zassert_equal((size_t)&NRF_EGU4->EVENTS_TRIGGERED[1],
+			  nrf_egu_event_address_get(NRF_EGU4, NRF_EGU_EVENT_TRIGGERED1),
+			  "NRF_EGU_4 incorrect address of event trigger 1");
+	zassert_equal((size_t)&NRF_EGU4->EVENTS_TRIGGERED[15],
+			  nrf_egu_event_address_get(NRF_EGU4,
+						NRF_EGU_EVENT_TRIGGERED15),
+			  "NRF_EGU_4 incorrect address of event trigger 15");
+
+	zassert_equal((size_t)&NRF_EGU5->EVENTS_TRIGGERED[0],
+			  nrf_egu_event_address_get(NRF_EGU5, NRF_EGU_EVENT_TRIGGERED0),
+			  "NRF_EGU_5 incorrect address of event trigger 0");
+	zassert_equal((size_t)&NRF_EGU5->EVENTS_TRIGGERED[1],
+			  nrf_egu_event_address_get(NRF_EGU5, NRF_EGU_EVENT_TRIGGERED1),
+			  "NRF_EGU_5 incorrect address of event trigger 1");
+	zassert_equal((size_t)&NRF_EGU5->EVENTS_TRIGGERED[15],
+			  nrf_egu_event_address_get(NRF_EGU5,
+						NRF_EGU_EVENT_TRIGGERED15),
+			  "NRF_EGU_5 incorrect address of event trigger 15");
+}
+
+void test_nrf_egu_channel_int_get(void)
+{
+	zassert_equal(NRF_EGU_INT_TRIGGERED0, nrf_egu_channel_int_get(0),
```

Comment:
Similar like in line 30, this bunch of code can be reduced to one for-loop and control by change number of elements in array